library/ssl_tls.c

```diff
@@ -1921,6 +1921,54 @@ int mbedtls_ssl_set_hs_ecjpake_password( mbedtls_ssl_context *ssl,
 
     return( 0 );
 }
+
+int mbedtls_ssl_set_hs_ecjpake_password_opaque( mbedtls_ssl_context *ssl,
+                                                mbedtls_svc_key_id_t pwd )
+{
+    psa_pake_cipher_suite_t cipher_suite = psa_pake_cipher_suite_init();
+    psa_pake_role_t psa_role;
+    psa_status_t status;
+
+    if( ssl->handshake == NULL || ssl->conf == NULL )
+        return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
+
+    if( ssl->conf->endpoint == MBEDTLS_SSL_IS_SERVER )
+        psa_role = PSA_PAKE_ROLE_SERVER;
+    else
+        psa_role = PSA_PAKE_ROLE_CLIENT;
+
+    if( mbedtls_svc_key_id_is_null( pwd ) )
+        return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
+    ssl->handshake->psa_pake_password = pwd;
+
+    psa_pake_cs_set_algorithm( &cipher_suite, PSA_ALG_JPAKE );
+    psa_pake_cs_set_primitive( &cipher_suite,
+                               PSA_PAKE_PRIMITIVE( PSA_PAKE_PRIMITIVE_TYPE_ECC,
+                                                   PSA_ECC_FAMILY_SECP_R1,
+                                                   256) );
+    psa_pake_cs_set_hash( &cipher_suite, PSA_ALG_SHA_256 );
+
+    status = psa_pake_setup( &ssl->handshake->psa_pake_ctx, &cipher_suite );
+    if( status != PSA_SUCCESS )
+        goto error;
+
+    status = psa_pake_set_role( &ssl->handshake->psa_pake_ctx, psa_role );
+    if( status != PSA_SUCCESS )
+        goto error;
+
+    psa_pake_set_password_key( &ssl->handshake->psa_pake_ctx,
```

Comment:
status is not checked here.

Reply:
How about adding a test that will exercise this failure? Passing a faulty pwd should work.

Reply:
`psa_pake_set_password_key` itself is tested, but the function you introduced - `mbedtls_ssl_set_hs_ecjpake_password_opaque` with a failing `psa_pake_set_password_key` was not :) 
In the previous state of this function, without saving the status returned by `psa_pake_set_password_key`, `mbedtls_ssl_set_hs_ecjpake_password_opaque` would return a success and some next call would probably fail. Or even would it? Or would we proceed without setting the password at all? It's just good practice to test failure points in introduced code if it's possible and not too time consuming / resource consuming / cumbersome.

Reply:
In theory yes, but in practice - we have to look at what data we're handling here and what can be "external" to the function that's added. `psa_role` is internal to the newly added function, and we can see that it's valid. `cipher_suite` is also internal to the function and if it works now, it will always work (or until someone changes it). The password, on the other hand, is external, and callers might put anything here, so a test is more justified.